### ISSUE TITLE
Regression: KotlinGenerator produces superfluous public keyword for overriding methods

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14883]
This is a cosmetic regression, introduced with a recent bugfix for:

- https://github.com/jOOQ/jOOQ/issues/14855

It can be seen in this commit, for example:
https://github.com/jOOQ/jOOQ-mcve/commit/e53f134144bf44ac48c88c3ab873f83b61a44528

```patch
diff --git a/jOOQ-mcve-kotlin-h2/src/main/kotlin/org/jooq/mcve/kotlin/h2/DefaultCatalog.kt b/jOOQ-mcve-kotlin-h2/src/main/kotlin/org/jooq/mcve/kotlin/h2/DefaultCatalog.kt
index a1cbd61..223c89a 100644
--- a/jOOQ-mcve-kotlin-h2/src/main/kotlin/org/jooq/mcve/kotlin/h2/DefaultCatalog.kt
+++ b/jOOQ-mcve-kotlin-h2/src/main/kotlin/org/jooq/mcve/kotlin/h2/DefaultCatalog.kt
@@ -29,7 +29,7 @@ open class DefaultCatalog : CatalogImpl("") {
      */
     val MCVE: Mcve get(): Mcve = Mcve.MCVE
 
-    override fun getSchemas(): List<Schema> = listOf(
+    public override fun getSchemas(): List<Schema> = listOf(
         Mcve.MCVE
     )
```

The fix in #14855 made sure that visibility cannot be *reduced*, but now, it's just always generated explicitly.

This regression is "cosmetic," meaning it does not produce any compilation errors, only a minor annoyance in generated code, e.g. when warnings are active for unnecessary keywords in the IDE.

**Comment by lukaseder** [comment]
There's a rare case where `public` is desired, e.g. when `-Xexplicit-api=strict` is specified

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.19.0, 3.18.3 (https://github.com/jOOQ/jOOQ/issues/14884), 3.17.12 (https://github.com/jOOQ/jOOQ/issues/14886), and 3.16.18 (https://github.com/jOOQ/jOOQ/issues/14885)
